src/utils/oauth2_util.py

```diff
@@ -428,21 +434,21 @@ def validate_roles(roles, secured_http_request_url):
 
 def validate_roles_and_scopes(secret_key, endpoint_request):
 
-    authorization_header = endpoint_request.headers.get("Authorization")
-    authorization_header_split = authorization_header.split(" ")
-    token = authorization_header_split[1]
-    token_decode = jwt.decode(token, secret_key, algorithms="HS256")
-    access_token_roles = token_decode.get("roles")
-    access_token_scopes = token_decode.get("scopes")
+    token = get_request_token(endpoint_request)
+
+    decoded_token = jwt.decode(token, secret_key, algorithms="HS256")
+
+    token_roles = decoded_token.get("user").get("roles")
```

Comment:
What if the user is None?

Reply:
Done. I used a empty dict as default value in case the key is not present in the dictionary
ex:
`token_roles = decoded_token.get("user", {}).get("roles")`